fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/BestMvSelector.java

```diff
@@ -50,10 +57,19 @@ public OptExpression selectBest() {
         for (OptExpression expression : expressions) {
             calculateStatistics(expression, context);
         }
+
+        // collect original table scans
+        List<Table> originalTables = MvUtils.getAllTables(queryPlan);
+        Set<ScalarOperator> predicates = MvUtils.getAllValidPredicates(queryPlan);
```

Comment:
Seems we can cache `originalTables ` and `predicates ` into MateralizationContext to avoid computing multi times for each rule apply.

Reply:
Here should get the tables and predicates of local QueryPlan, not the wholly origin query plan.